plugins/topology/README.md

```diff
@@ -190,44 +207,46 @@ The Topology plugin enables you to visualize the workloads such as Deployment, J
+  To retrieve the instance URL, CheCluster Custom Resource (CR) is required. The instance URL is not retrieved if the namespace annotation value is different from `openshift-devspaces` as CheCluster CR is created in `openshift-devspaces` namespace.
 
-- The following annotation is added to the entity's `catalog-info.yaml` file to identify whether an entitiy contains the Kubernetes resources:
+- A custom label selector is added, which Backstage uses to find the Kubernetes resources. The label selector takes precedence over the ID annotations.
 
   ```yaml title="catalog-info.yaml"
   annotations:
-    backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
+    backstage.io/kubernetes-label-selector: 'app=my-app,component=front-end'
   ```
 
-  You can also add the `backstage.io/kubernetes-namespace` annotation to identify the Kubernetes resources using the defined namespace.
+  If you have multiple entities while Red Hat Dev Spaces is configured and want multiple entities to be able to support the edit code decorator that redirects to the Red Hat Dev Spaces instance. You can add the `backstage.io/kubernetes-label-selector` annotation to the `catalog-info.yaml` file for each entity.
 
   ```yaml title="catalog-info.yaml"
   annotations:
-    backstage.io/kubernetes-namespace: <RESOURCE_NS>
+    backstage.io/kubernetes-label-selector: 'component in (<BACKSTAGE_ENTITY_NAME>,che)'
   ```
 
-  > The Red Hat OpenShift Dev Spaces instance is not accessible using the edit code decorator if the `backstage.io/kubernetes-namespace` annotation is added to the `catalog-info.yaml` file.
-
-  To retrieve the instance URL, CheCluster Custom Resource (CR) is required. The instance URL is not retrieved if the namespace annotation value is different from `openshift-devspaces` as CheCluster CR is created in `openshift-devspaces` namespace.
+  If you are using the previous label selector, ensure that you add the following labels to your resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:
 
-- A custom label selector is added, which Backstage uses to find the Kubernetes resources. The label selector takes precedence over the ID annotations.
+  ```yaml
+  labels: component=<BACKSTAGE_ENTITY_NAME>
+  ```
 
-  ```yaml title="catalog-info.yaml"
-  annotations:
-    backstage.io/kubernetes-label-selector: 'app=my-app,component=front-end'
+  ```yaml title="checluster.yaml"
+  labels: component=che
   ```
 
+  You can also write your own custom query for the label selector with unique labels to differentiate your entities. However, you need to ensure that you add those labels to the resources associated with your entities including your CheCluster instance.
+
 - The following label is added to the resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:
 
-  ```yaml title="catalog-info.yaml"
+  ```yaml
   labels:
     backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>`
   ```
```

Comment:
this seems out of context to me, should this be with the annotations block or same https://github.com/janus-idp/backstage-plugins/pull/730/files#diff-53fc015c5ba5c543df0eb8f8fafb3604ef2b59b0c27c6d37ce3de17f8d0feeb7R158-R163